### ISSUE TITLE
a11y: keyboard navigation performing incorrect action on resources list

### ISSUE THREAD

**Issue by IsaSih** [rancher/dashboard#13239]
**Setup**
Rancher version:
Rancher | v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
Dashboard | master https://github.com/rancher/dashboard/commit/6eaaa6aa9cceca28dea5385bfea4de228ce09c49
Rancher CLI | v2.11.0-rc.1

**Describe the bug**
In pages with list of resources, in this example, the cluster dashboard, when the user navigates to the certificates tab using the keyboard navigation, then hit enter/return key to select the checkbox of a certificate in the table, instead of the resource being selected, the user is redirected to the secrets page. Performing the same action with the space bar, simply selects the resource, as expected.

**To Reproduce**
In the cluster dashboard page, navigate to the certificates tab using the keyboard navigation, then use enter/return key to select the checkbox of a certificate in the table. Observe that the UI redirects the user to another page.

**Result**
The enter/return key to select the checkbox of a certificate in the table, redirects the user to another page instead of performing the action to select the resource.

**Expected Result**
The enter/return key must select the checkbox of a given resource.

**Screenshots**

https://github.com/user-attachments/assets/e0e15453-de03-4138-9abf-efbc7472378b

**Comment by aalves08** [comment]
After quite a bit of debugging, the problem lies in this `keyup` event handler, which is fired after you press enter on the checkbox: https://github.com/rancher/dashboard/blob/master/shell/components/ResourceTable.vue#L562-L566

This is why pressing `enter` on the checkbox in a table row opens the details view...

Original work: https://github.com/rancher/dashboard/issues/7048
Follow-up work because of bug: https://github.com/rancher/dashboard/issues/8094

We are reaching a point of events propagation nightmare right now because of stuff like this.

What do you think should be the path forward @rak-phillip ? I wasn't able stop propagation/preventDefault on the checkbox side 🤔

**Comment by rak-phillip** [comment]
@aalves08 the issue is two-fold:

1. The event listener for then enter key is on the window, making the row selection a global behavior when a ResourceTable is rendered
2. The design of the Checkbox component doesn't lend itself well to common patterns to stop event propagation

**Comment by IsaSih** [comment]
Tests pass in 
Rancher | v2.11-3d729db8a848fb2523bbc2f3f21aa5367dcdcc69-head
Dashboard | master df18acc
Rancher CLI | v2.11.0-rc.1

https://github.com/user-attachments/assets/06cebcd2-e1af-4e37-88d3-c098113f56db
